multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -415,20 +406,28 @@ func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
 	klog.InfoS("Starting StaleResCleanupController")
 	defer klog.InfoS("Shutting down StaleResCleanupController")
 
-	ctx, _ := wait.ContextForChannel(stopCh)
-	if err := c.RunOnce(ctx); err != nil {
-		c.Enqueue()
-		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	ctx, cancel := wait.ContextForChannel(stopCh)
+
+	switch c.clusterRole {
+	case LeaderCluster:
+		go wait.UntilWithContext(ctx, c.cleanupMemberClusterAnnounces, memberClusterAnnounceStaleTime/2)
+	case MemberCluster:
+		if err := c.RunOnce(ctx); err != nil {
```

Comment:
Do we still need `RunOnce()` or just do `runWorker()` here too?

Reply:
runWorker is for handling the queue event. But you are right, the RunOnce() is not longer needed, I replaced it with `CleanUp()`.